src/mame/atari/atari400.cpp

```diff
@@ -2057,6 +1858,25 @@ uint8_t a400_state::djoy_b_r()
 	return b;
 }
 
+uint8_t a1200xl_state::djoy_b_r()
+{
+	uint8_t b = 0;
+	for (int i = 0; i < 2; i++)
+		if (!m_ctrl[i].found() || BIT(m_ctrl[i]->read_joy(), 5))
+			b |= 1 << i;
+
+	// TODO: TRIG2 (0) for XEGS keyboard disconnected, always 1 otherwise
+	b |= 1 << 2;
+	// TODO: TRIG3 should really read from RD5 instead
+	// - non-64KB SDX boot (will hang in kernel during POST by boot conflict with BASIC if not handled properly)
+	// - a1200xl cart boot (expects this to be reversed)
+	// - returning m_cartleft->exists() makes BASIC disable thru OPTION to not work and makes maxflash to fail loading most .xex
+//  b |= m_cart_rd5_enabled << 3;
+	b |= !m_cartleft->exists() << 3;
```

Comment:
This is unpleasant and annoying, I really wonder now if RD5 initialization already requires a non-zero frame startup ...